src/netex_idf/offers.rs

```diff
@@ -234,6 +235,26 @@ where
     Ok(routes)
 }
 
+fn update_validity_period(
```

Comment:
This function is basically a copy of what is done in the TransXChange connector. I think it would be nice to extract these functions into a common part instead of duplicating code.